### ISSUE TITLE
400 instead of 500 on error of STREAM.MSG.GET requests

### ISSUE THREAD

**Issue by scottf** [nats-io/nats-server#1976]
## Feature Request
When making a STREAM.MSG.GET request with an invalid sequence respond with a 400 error code instead of a 500 error code. Currently, the response has error code 500, which suggests a server error.

Also, maybe combine the use cases into the same error as it's irrelevant to the client why the message is not found.

#### Use Case:

The first can happen when the requested sequence is 0 or for a message that was deleted. The second is when the requested sequence is greater than the last actual sequence number.

```
{"type":"io.nats.jetstream.api.v1.stream_msg_get_response","error":{"code":500,"description":"no message found"}}
{"type":"io.nats.jetstream.api.v1.stream_msg_get_response","error":{"code":500,"description":"stream store EOF"}}
```

#### Who Benefits From The Change(s)?
More standard error code, where 4xx represents a user/input error instead of 5xx representing a server error.

**Comment by ripienaar** [comment]
Agree, should be 404

```
17:47:37 >>> $JS.API.STREAM.MSG.GET.CHORIA_REGISTRATION
{"seq":10}

17:47:37 <<< $JS.API.STREAM.MSG.GET.CHORIA_REGISTRATION
{"type":"io.nats.jetstream.api.v1.stream_msg_get_response","error":{"code":500,"description":"no message found"}}
```

**Comment by derekcollison** [comment]
This with no wait set or no wait is not set? Batch size?

**Comment by ripienaar** [comment]
It’s the stream get APi not consumer pull. 

That api does a message lookup and return 500 for all errors - msg not found is a error from the storage

**Comment by derekcollison** [comment]
Ah gotcha.. ok that makes more sense ;)

**Comment by kozlovic** [comment]
Closing this since it is addressed with #2053
